modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ObjectRelationship/ObjectRelationship.js

```diff
@@ -12,46 +12,133 @@
  * details.
  */
 
-import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocomplete';
+import ClayAutocomplete from '@clayui/autocomplete';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import {useDebounce} from '@clayui/shared';
 import {FieldBase} from 'dynamic-data-mapping-form-field-type/FieldBase/ReactFieldBase.es';
+import {fetch} from 'frontend-js-web';
 import React from 'react';
 
-const ObjectRelationship = ({
+const NETWORK_STATUS_LOADING = 1;
+const NETWORK_STATUS_UNUSED = 4;
+
+const LoadingWithDebounce = ({loading, render}) => {
+	const debouncedLoadingChange = useDebounce(loading, 500);
+
+	if (loading || debouncedLoadingChange) {
+		return (
+			<ClayDropDown.Item className="disabled">
+				{Liferay.Language.get('loading')}
+			</ClayDropDown.Item>
+		);
+	}
+
+	return render;
+};
+
+export function ObjectRelationship({
 	apiURL,
+	inputName,
+	labelKey = 'label',
 	initialLabel,
 	initialValue,
-	inputName,
-	itemsKey,
-	itemsLabel,
 	name,
+	onBlur = () => {},
 	onChange,
-	placeholder,
+	onFocus = () => {},
+	placeholder = Liferay.Language.get('search'),
+	predefinedValue,
 	readOnly,
 	required,
 	value,
+	valueKey = 'value',
 	...otherProps
-}) => (
-	<FieldBase
-		name={name}
-		readOnly={readOnly}
-		required={required}
-		{...otherProps}
-	>
-		<Autocomplete
-			{...(placeholder && {inputPlaceholder: placeholder})}
-			apiUrl={apiURL}
-			initialLabel={initialLabel}
-			initialValue={initialValue}
-			inputName={inputName}
-			itemsKey={itemsKey}
-			itemsLabel={itemsLabel}
+}) {
+	const mutatedRef = React.useRef(false);
```

Comment:
Just a question about it, is this to help you switch when to use `value` or initial/predefined value?

Reply:
Yeah this was to make sure the initial/predefined is only there when the input hasn't been modified. Since the onChange and value come from outside this component, we aren't able to pass it as an initial state.

Otherwise if we do something like `value={value || predefinedValue}` it would get weird if the user clears the input and then it would render the predefined again.

Reply:
Hmm yeah makes sense, a good strategy.